### ISSUE TITLE
onActivateTick can cause infinite loops

### ISSUE THREAD

**Issue by j2L4e** [scttcper/ngx-toastr#226]
My use case is a custom error handler showing toasts:

```
export class ToastErrorHandler extends ErrorHandler {
  constructor(private injector: Injector) {
    super();
  }

  handleError(error: Error): void {
    const toastr = this.injector.get(ToastrService);

    const config: Partial<IndividualConfig> = {
      timeOut: 10000
    };

    toastr.error(error.message, config);

    super.handleError(error);
  }
}
```

Obviously this toast won't show as expected, because the error handler runs outside angular's zone (for good reason).
Using `onActivateTick` seems tempting, but `tick()` triggers change detection on the entire component tree. If the handled error is raised during change detection, it will be raised again and again and freeze the browser tab.

I'm pretty sure that calling `appRef.tick()` is overkill in any scenario, because what we actually need is change detection inside the toast: `toastr.error('blah').portal.changeDetectorRef.detectChanges();`

Toasts should run CD on themselves when necessary instead of relying on ticks.

**Comment by yarrgh** [comment]
I'm not able duplicate this where the custom error handler triggers outside of change detection. Every attempt I make triggers change control except for triggering an error out side of  Angular's zone (which seems to bypass the error handler altogether). I'm not using the `activateOnTick` config option.

Can you give me an easy code sample I can run that will trigger the ErrorHandler to run outside of change detection?

**Comment by yarrgh** [comment]
I think I just found a way by triggering an error inside a component's template.

**Comment by scttcper** [comment]
Published 7.0.0-beta.0 on ngx-toastr@next
